File: 04.SERVIDOR/WS_ComElect_RESTJAVA_GR05/src/main/java/ec/edu/yaranga/comerelect/dto/FacturaResponse.java
package ec.edu.yaranga.comerelect.dto;

import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

@Data
public class FacturaResponse {
    private String numFactura;
    private LocalDate fecha;
    private String cedula;
    private String nombreCliente;
    private BigDecimal total;
    private String tipoPago;
    private BigDecimal descuento;
    private List<DetalleDTO> detalles;

    @Data
    public static class DetalleDTO {
        private String codigo;
        private String nombre;
        private Integer cantidad;
        private BigDecimal precio;
        private BigDecimal subtotal;
    }
}

File: 04.CLI MOV/CLIMOV_ComerElect_RESTJAVA_GR05/app/src/main/java/ec/edu/monster/modelo/FacturaResponse.kt
package ec.edu.monster.modelo

import java.math.BigDecimal
import java.time.LocalDate

data class FacturaResponse(
    val numFactura: String,
    val fecha: LocalDate,
    val cedula: String,
    val total: BigDecimal,
    val tipoPago: String,
    val descuento: BigDecimal,
    val detalles: List<DetalleDTO>
) {
    data class DetalleDTO(
        val codigo: String,
        val nombre: String,
        val cantidad: Int,
        val precio: BigDecimal,
        val subtotal: BigDecimal
    )
}
